unity/src/test/scala/io/delta/unity/UCCatalogManagedCommitterSuite.scala

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.unity
+
+import java.util.Optional
+
+import scala.collection.JavaConverters._
+
+import io.delta.kernel.commit.CommitMetadata
+import io.delta.kernel.internal.tablefeatures.TableFeatures
+import io.delta.kernel.test.VectorTestUtils
+
+import org.scalatest.funsuite.AnyFunSuite
+
+class UCCatalogManagedCommitterSuite
+    extends AnyFunSuite
+    with UCCatalogManagedTestUtils
+    with VectorTestUtils {
+
+  test("constructor throws on null inputs") {
+    val ucClient = new InMemoryUCClient("ucMetastoreId")
+
+    assertThrows[NullPointerException] {
+      new UCCatalogManagedCommitter(null, "ucTableId", baseTestTablePath)
+    }
+    assertThrows[NullPointerException] {
+      new UCCatalogManagedCommitter(ucClient, null, baseTestTablePath)
+    }
+    assertThrows[NullPointerException] {
+      new UCCatalogManagedCommitter(ucClient, "ucTableId", null)
+    }
+  }
+
+  test("commit throws on null inputs") {
+    val ucClient = new InMemoryUCClient("ucMetastoreId")
+    val committer = new UCCatalogManagedCommitter(ucClient, "ucTableId", baseTestTablePath)
+
+    // Null engine
+    assertThrows[NullPointerException] {
+      committer.commit(null, emptyActionsIterator, catalogManagedWriteCommitMetadata())
+    }
+
+    // Null finalizedActions
+    assertThrows[NullPointerException] {
+      committer.commit(defaultEngine, null, catalogManagedWriteCommitMetadata())
+    }
+
+    // Null commitMetadata
+    assertThrows[NullPointerException] {
+      committer.commit(defaultEngine, emptyActionsIterator, null)
+    }
+  }
+
+  test("commit throws if CommitMetadata is for a different table") {
```

Comment:
check that this does normalize the path?

Reply:
yeah this is interesting.

(a) I think our UCCatalogManagedClient should be resolving the `tablePath` input using the engine
(b) all I want to do in the committer is make sure the input delta log path belongs to that tablePath. think I can just compare the paths directly? normalizing to URI seems also fine to me ... but I'm not expecting any of the input to be weird/bad...

Reply:
But part of me wonders: if the catalog told us that X is the table path, should the engine then override and "resolve" it (and maybe change it slightly?) hmmm

Reply:
My opinion: The UCCatalogManagedClient should resolve the input path ... that's how everything inside of Kernel operates.

Then in this class: we can just use `Path::getParent`

Reply:
cc @vkorukanti or @huan233usc if you have any thoughts?

Reply:
Alas -- my updated opinoin is:

Let's just keep things simple. The engine/catalog should give the UCCatalogManagedClient the right tablePath. We should just use that. We can iterate and improve on this later